pywwt/annotation.py

```diff
@@ -252,6 +255,218 @@ def _on_trait_change(self, changed):
+                panels = self.dim[telescope][1]
+                
+                side = (instr_h * u.arcsec).to(u.deg).value
+                mid = [(instr_h * u.arcsec).to(u.deg).value / 2.,
+                       (instr_w * u.arcsec).to(u.deg).value / 2.]
+                
+                gap = (1.52 * u.arcmin).to(u.deg).value / 2.
+
+                y_panel = side + gap
+                i = 0
+                while(i < panels):
+                    corners = self._rotate(ra, dec + y_panel, mid, origin, rot)
+                    annot = self.parent.add_polygon(corners, **kwargs)
+                    self.active.append(annot)
+
+                    y_panel -= side + gap
+                    i += 1
+        else:
+            raise ValueError('the given telescope\'s field of view is unavailable at this time')
+
+    def _rotate(self, ra, dec, mid, origin, rot):
+        # convert rot to radians for numpy's trig functions
+        rot = rot.to(u.rad).value
+        cos = np.cos(rot); sin = np.sin(rot)
+        
+        # if the point at the center of rotation is not the center of the shape:
+        o_ra = origin[0]; o_dec = origin[1]
+
+        # get coord values of corners of the shape...
+        # by subtracting half the length of a side
+        # from bottom left counter-clockwise: (+-, --, -+, ++) (--, +-, ++, -+)
+        ra_mn = ra - mid[1]; ra_pl = ra + mid[1]
+        dec_mn = dec - mid[0]; dec_pl = dec + mid[0]
+
+        # check that abs(dec) < 90;
+        # if not, adjust and set the corresponding ra to be swapped by 180
+        swap0 = 0.; swap1 = 0.
+        if abs(dec_mn) > 90:
+            # must check both cases for each dec b/c some panels
+            # can be entirely above/below a pole (e.g. in Spitzer),
+            # in which case abs(dec) > 90 for both decs
+            if dec_mn < -90:
+                dec_mn = -90. - (dec_mn + 90.)
+            else: # dec_mn > 90
+                dec_mn = 90. - (dec_mn - 90.)
+            swap0 = 180.
+        if abs(dec_pl) > 90:
+            if dec_pl > 90:
+                dec_pl = 90. - (dec_pl - 90.)
+            else: # dec_pl < -90
+                dec_pl = -90. - (dec_mn + 90.)
+            swap1 = 180.
+        
+        # temporarily translate corners pre-rotation
+        r0 = ra_mn - o_ra; r1 = ra_pl - o_ra
+        d0 = dec_mn - o_dec; d1 = dec_pl - o_dec
+
+        # then, use trigonometry to rotate and undo the previous translation
+        rotated = concatenate((
+            SkyCoord(o_ra + r1*cos - d0*sin + swap0,
+                     o_dec + r1*sin + d0*cos, unit='deg'),
+            SkyCoord(o_ra + r0*cos - d0*sin + swap0,
+                     o_dec + r0*sin + d0*cos, unit='deg'),
+            SkyCoord(o_ra + r0*cos - d1*sin + swap1,
+                     o_dec + r0*sin + d1*cos, unit='deg'),
+            SkyCoord(o_ra + r1*cos - d1*sin + swap1,
+                     o_dec + r1*sin + d1*cos, unit='deg')))
```

Comment:
@ojustino - here the offsets applied to RA should be ``/ cos(o_dec)``.

Reply:
Do you mean `o_ra` alone or the whole RA term?

Reply:
I meant it should be ``o_ra + (r1*cos - d1*sin + swap1) / cos(o_dec)`` but in fact this code might be changed because of my comment below anyway.